### ISSUE TITLE
The loss value calculated by reevaluation seems incorrect using subclass of keras model.

### ISSUE THREAD

**Issue by yuyicg** [sql-machine-learning/elasticdl#617]
Using subclass of keras model, the loss value of mnist dataset calculated by reevaluation is about 1.7 while the final loss value in training job is less than 1.
Here are commands to reproduce this issue:
- Submit the training job and wait the job to finished.
```bash
python elasticdl/python/elasticdl/client/client.py \
    --job_type=training \
    --job_name=training-job \
    --model_file=elasticdl/python/examples/mnist_subclass.py \
    --training_data_dir=/data/mnist_nfs/mnist/train \
    --evaluation_data_dir=/data/mnist_nfs/mnist/test \
    --master_pod_priority=high-priority \
    --worker_pod_priority=high-priority \
    --num_epoch=1 \
    --minibatch_size=64 \
    --records_per_task=100 \
    --num_workers=1 \
    --master_cpu_request=1000m \
    --master_cpu_limit=1000m \
    --master_memory_request=1024Mi \
    --master_memory_limit=1024Mi \
    --worker_cpu_request=3000m \
    --worker_cpu_limit=3000m \
    --worker_memory_request=4096Mi \
    --worker_memory_limit=4096Mi \
    --grads_to_wait=1 \
    --codec_type=bytes \
    --mount_path=/data \
    --volume_name=data-volume \
    --repository=gcr.io \
    --image_base=gcr.io/elasticdl/elasticdl:dev \
    --image_pull_policy=Always \
    --checkpoint_dir=/data/cp \
    --checkpoint_step=10 \
    --keep_checkpoint_max=1 \
    --log_level=INFO
```
- Submit the evaluation job and see the loss from logs of the eval pod.
```bash
python elasticdl/python/elasticdl/client/client.py \
    --job_type=evaluation \
    --job_name=eval-job \
    --model_file=elasticdl/python/examples/mnist_subclass.py \
    --trained_model=/data/cp/model_v1200.chkpt \
    --data_dir=/data/mnist_nfs/mnist/train \
    --minibatch_size=64 \
    --eval_pod_priority=high-priority \
    --eval_cpu_request=1000m \
    --eval_cpu_limit=1000m \
    --eval_memory_request=1024Mi \
    --eval_memory_limit=1024Mi \
    --codec_type=bytes \
    --repository=gcr.io \
    --mount_path=/data \
    --volume_name=data-volume \
    --image_base=gcr.io/elasticdl/elasticdl:dev \
    --image_pull_policy=Always \
    --log_level=INFO
```
